iceoryx_hoofs/test/moduletests/test_console_logger.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/log/platform_building_blocks/console_logger.hpp"
+
+#include "iceoryx_hoofs/log/logging.hpp"
+#include "iceoryx_platform/fcntl.hpp"
+#include "test.hpp"
+
+#include <cstdio>
+#include <iostream>
+
+namespace
+{
+using namespace ::testing;
+
+class LoggerSUT : public iox::pbb::ConsoleLogger
+{
+  public:
+    using iox::pbb::ConsoleLogger::flush;
+    using iox::pbb::ConsoleLogger::logString;
+};
+
+#if !defined(_WIN32)
+TEST(ConsoleLogger_test, TestOutput)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "67f1dac5-b425-414a-9690-268ecb06c1ee");
+    GTEST_SKIP() << "todo iox-#1345 this does not work on the CI and needs to be refactored. An alternative would be "
+                    "to run a binary which uses the logger and capture its output.";
+
+    constexpr const char* LOG_FILE_NAME{"iceoryx_console_logger_output_test.txt"};
+    constexpr const char* LOG_MESSAGE{"All glory to the hypnotoad!"};
+
+    // #### redirect stdout ####
+    // to be able to check if the logger writes to stdout we need to redirect the output to a file
+    // save original stdout
+    auto stdoutOriginal = dup(STDOUT_FILENO);
+
+    int logFd = open(LOG_FILE_NAME, O_CREAT | O_WRONLY | O_TRUNC, 0664);
```

Comment:
This might work for Windows, but I doubt it since the file descriptor is duplicated and we don't have an implementation for `dup2`. It also wouldn't solve the problem in the CI though.
The proper solution is mentioned in the test following GTEST_SKIP. I wouldn't waste time for a solution we can't use anyway. I kept the code just in case the proposed solution does not work and we invest time to make the current one work in the CI. Are you okay with commenting the the whole code in the test out?

Reply:
According to this it is deprecated but still works: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/posix-dup-dup2?view=msvc-170

I think also that `dup` and `dup2` are also easy implemented since the manual suggest that one should use now `_dup` and `_dup2` with the same interface.

Just to be clear I did not overlook something. We just disable the test for windows but the functionality should in theory be the same on windows as it is on linux (for the logger)?

Reply:
Yes it is currently compile time disabled for windows and runtime disabled for all platforms since the approach does not work in the CI.

Reply:
After some thinking, I think the test can be removed when an launch testing integration test checks for the output. This would save us from a hard to maintain test

Reply:
@MatthiasKillat the new order of @dkroenke is to use `GTEST_SKIP` instead of the `DISABLE_` prefix ;)